assets/src/hooks/useSearchResults.ts

```diff
@@ -33,4 +34,54 @@ const useSearchResults = (
   })
 }
 
+const limitedSearchResultsData = type({
+  matching_vehicles: dataStruct,
+  has_more_matches: boolean(),
+})
+
+type LimitedSearchResultsData = Infer<typeof limitedSearchResultsData>
+export type LimitedSearchResults = {
+  matchingVehicles: (Vehicle | Ghost)[]
+  hasMoreMatches: boolean
+}
+
+const parseLimitedSearchResults = (
+  data: LimitedSearchResultsData
+): LimitedSearchResults => ({
+  matchingVehicles: parser(data.matching_vehicles),
+  hasMoreMatches: data.has_more_matches,
+})
+
+export const useLimitedSearchResults = (
+  socket: Socket | undefined,
+  query: { property: SearchProperty; text: string; limit: number } | null
+): LimitedSearchResults | null => {
+  const topic: string | null =
+    query && `vehicles_search:limited:${query.property}:${query.text}`
+  const [state, pushUpdate] = useCheckedTwoWayChannel<
+    LimitedSearchResultsData,
+    LimitedSearchResults | null
+  >({
+    socket,
+    topic,
+    event: "search",
+    dataStruct: limitedSearchResultsData,
+    parser: parseLimitedSearchResults,
+    loadingState: null,
+    initialMessage: query && {
+      limit: query.limit,
+    },
+  })
+
+  useEffect(() => {
+    if (query) {
+      pushUpdate("update_search_query", {
+        limit: query.limit,
+      })
+    }
+  }, [query?.property, query?.text, query?.limit, pushUpdate])
```

Comment:
**suggestion:** note that the React hooks linting rules still don't like this and instead need `query` itself to be a dependency if it's used. To prevent excessive re-renders you can take the approach from some other hooks like `useSwings` where you have a separate state that contains the query, then deep compare the query passed in as an argument to the saved query and only update the state if the deep comparison reveals a difference.

Reply:
Thanks, I'll take a look at `useSwings`! There is a similar hook issue related to `initialMessage` being passed to the channel on join. Will try to do something crafty there so that the value for `initialMessage` is only updated when relevant.